### PR TITLE
Beginning API integration

### PR DESCRIPTION
**Pull request by GomeChas**
I'm beginning some of the integration with the db.

- Added several sprocs to populate UI form and tables
- added several new data models
- properly split category out of posting table and into separate entity
- added some routs on the Flask API to access db by UI
- Ingested description now includes the `.encode().decode()` fix that was otherwise being used by the fetch script

### CODE REVIEW

File: backend/data/Job.db
Comment (cdelapena) [inline backend/data/Job.db:1]: Needed to delete Job.db and recreate by running jobs_ingestion.py to include the preprocessed_description column. Recommend PR excludes Job.db.

File: backend/sprocs.py [hunk 1]
```diff
@@ -146,9 +145,105 @@ def get_job_postings_by_category(categories: str, db_filename: str) -> str:
...
+    """
+    items = [item.strip() for item in categories.split(",")]
+    category_req = ", ".join(f"'{item}'" for item in items)
+
+    query = f"""
+        SELECT
+            p.id,
+            c.name,
+            job_title,
+            description,
```
Comment (cdelapena) [inline backend/sprocs.py:173]: For model, recommend replacing description with preprocessed_description.

**Review by cdelapena — COMMENTED** [review]
Really great work! Just a small number adjustments.

File: backend/app.py [hunk 2]
```diff
@@ -1,15 +1,41 @@
+from flask import Flask, make_response, jsonify
 from fetch_external_data import return_clean_json_data
 from text_preprocessing import preprocess_text
```
[the remark this answers is not in the document]
Comment (GomeChas) [reply backend/app.py]: Good call-out.  Adjusted the import.

File: backend/utils/sql/sql.py [hunk 3]
```diff
@@ -216,7 +274,10 @@ def db_ingestion(df: pd.DataFrame, db_filename: str = "Job.db") -> None:
         print(f"Inserting [{db_filename.split('.')[0]}].[sources] table...")
         insert_new_sources(df, jobs_conn)
 
-        print(f"Inserting [{db_filename.split('.')[0]}].[tags] table...")
+        print(f"Inserting [{db_filename.split(".")[0]}].[categories] table...")
```
[the remark this answers is not in the document]
Comment (GomeChas) [reply backend/utils/sql/sql.py]: wow, I shoulda caught that!

File: backend/utils/sql/models.py [hunk 4]
```diff
@@ -12,4 +12,22 @@ class JobPosting:
...
     source_url: str
-    publish_date: str
\ No newline at end of file
+    publish_date: str
+
+@dataclass
+class ModelData:
+    id: int
+    job_title: str
+    description: str
```
[the remark this answers is not in the document]
Comment (GomeChas) [reply backend/utils/sql/models.py]: correct for both the getRecord sprocs and the getModelData sprocs.

File: backend/utils/sql/sql.py [hunk 5]
```diff
@@ -216,7 +274,10 @@ def db_ingestion(df: pd.DataFrame, db_filename: str = "Job.db") -> None:
...
+        insert_new_categories(df, jobs_conn)
+
+        print(f"Inserting [{db_filename.split(".")[0]}].[tags] table...")
```
[the remark this answers is not in the document]
Comment (GomeChas) [reply backend/utils/sql/sql.py]: amended

File: backend/data/Job.db
Comment (GomeChas) [reply backend/data/Job.db:1]: added Job.db to .gitignore.  Too large to make work anyway and we have support to build, and ingest as needed.

File: backend/sprocs.py (hunk 1 above)
Comment (GomeChas) [reply backend/sprocs.py]: sprocs for display in the UI ought to use the html cleaned, but not preprocessed description.  This has been adjusted.

**Review by cdelapena — APPROVED** [review]
Looks great! Reads clean and runs when tested. Approved.

**Pull request merged**